share/spice/libraries/libraries.js

```diff
@@ -0,0 +1,207 @@
+            signal = "high";
+        } else if(api_result[0].platform === "Go" && pkgname.split("/").pop() == query) {
+            template = text_template;
+            api_result = api_result[0];
+            signal = "high";
+        } else {
+            template = tile_template;
+            signal = "low";
+            meta.sourceName = 'libraries.io';
+            meta.sourceUrl = 'https://libraries.io/api/search?q=' + search_items;
+        }
+
+        Spice.add({
+            id: 'libraries',
+            name: 'Libraries',
+            signal: signal,
+            data: api_result,
+            meta: meta,
+            templates: template,
+            sort_fields: {
+                rank: function(a, b) {
+                    return a.rank > b.rank ? -1 : 1;
+                }
+            },
+            sort_default: 'rank',
+            normalize: function(item) {
+
+                console.log(item.name);
+
+                var infoboxData = [
+                    { heading: 'Package information:' },
+                ];
+
+                // show the license if it exists
+                if(item.latest_release_number !== undefined 
+                    && item.latest_release_number !== "" 
+                    && item.latest_release_number !== null) {
+
+                    infoboxData.push({
+                        label: "Version",
+                        value: item.latest_release_number,
+                    });
+                }
+
+                // show the license type if it exists
+                if(item.normalized_licenses[0] !== undefined) {
+                    infoboxData.push({
+                        label: "License",
+                        value: item.normalized_licenses[0],
+                    });
+                }
+
+                // show stars if there is any
+                if(item.stars > 0) {
+                    infoboxData.push({
+                        label: "Stars",
+                        value: item.stars,
+                    });
+                }
+
+                // show forks if there is any
+                if(item.forks > 0) {
+                    infoboxData.push({
+                        label: "Forks",
+                        value: item.forks,
+                    });
+                }
+
+                var package_manager = item.platform;
+                var url;
+
+                if(item.homepage != null && item.homepage != "") {
+                    url = item.homepage;
+                } else if(item.package_manager_url != null && item.package_manager_url != "") {
+                    url = item.package_manager_url;
+                } else if(item.repository_url != null && item.homepage != item.repository_url) {
+                    url = item.repository_url;
+                } else {
+                    url = 'https://libraries.io/api/search?q=' + search_items;
+                }
+
+                if (item.name && item.description) {
```

Comment:
Should we bail earlier if these are not true?

i.e. at the top `if (!item.name && item.description) { return }`

Alternatively, you could set a relevancy block to require these to be defined. You'd need to set the relevancy block depending on whether we're doing a list or detail view